### ISSUE TITLE
decodeBuffer() might be leaking event listeners

### ISSUE THREAD

**Issue by mikethea1** [hoppergee/heic-to#3]
It looks like decodeBuffer is adding an event listener to the worker singleton on every call but never removing it. Would it be worth changing the code to something like this?

```
const decodeBuffer = async (buffer) => {
  return new Promise((resolve, reject) => {
    const id = (Math.random() * new Date().getTime()).toString();
    const message = { id, buffer };

    const listener = message => {
      if (message.data.id === id) {
        worker.removeEventListener("message", listener); // Remove the listener
        if (message.data.error) {
          return reject(message.data.error);
        }
        return resolve(message.data.imageData);
      }
    };

    worker.addEventListener("message", listener);
    worker.postMessage(message);
  });
};
```

**Comment by hoppergee** [comment]
Thank you @mikethea1  I'll take a look.

**Comment by hoppergee** [comment]
@mikethea1 I just publish a fix with [v1.0.3](https://github.com/hoppergee/heic-to/releases/tag/v1.0.3). Please have a test.

**Comment by mikethea1** [comment]
Thanks @hoppergee ! I just tested 1.1.5 (which I assume also has the fix) and everything seems to be working well.
